openedx/openedx-platform
# feat: Adds additional data and filters to discussions API [BD-38] [TNL-7617] [BB-2969]

This change adds three new filters to the threads API. They are:
* Filtering only threads that are flagged for abuse
* Filtering by the thread type (discussion or question)
* Filtering by the thread author

In addition, it also adds a new ``abuse_flagged_count`` field for threads. It
returns a count of the number of comments in a thread that are flagged for abuse.
This is only visible to users that have moderator privileges or higher.

Finally, it also adds a ``abuse_flagged_any_user`` field that is set if any user
has flagged a thread. This field, too, is only visible to moderators or above.

**JIRA tickets**: TNL-7617

**Dependencies**: https://github.com/edx/cs_comments_service/pull/328

**Sandbox URL**: 
    LMS: https://pr27914.sandbox.opencraft.hosting/
    Studio: https://studio.pr27914.sandbox.opencraft.hosting/

**Testing instructions**:

1. Get an Oauth/JWT token
2. Create some discussion data with multiple users, using anonymous posts, and question, and discussion posts, and flag some comments
3. Access the threads api at http://localhost:18000/api/discussion/v1/threads/?course_id={course_id} and check for the following
  - Specifying an author different from the current user should return all non-anonymous threads by that author.
  - Specifying an author that is the same as the logged in user should return all their threads. 
  - Setting `flagged` to true should return all flagged threads
  - Specify 'thread_type` as either question or discussion and see only those types of threads are returned
  - Check that for a staff user, the API should return the count of abuse flagged comments inside a thread, and also return if a comment has been flagged by any user, not just the current user. 
  - Check that fir a non-staff user the above aren't returned. 

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
